relay-profiling/src/sample.rs

```diff
@@ -0,0 +1,426 @@
+use std::collections::HashMap;
+
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+
+use relay_general::protocol::{Addr, EventId};
+
+use crate::error::ProfileError;
+use crate::native_debug_image::NativeDebugImage;
+use crate::transaction_metadata::TransactionMetadata;
+use crate::utils::deserialize_number_from_string;
+use crate::Platform;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct Frame {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    instruction_addr: Option<Addr>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    name: Option<String>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    line: Option<u32>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    file: Option<String>,
```

Comment:
Since all the attributes are optional, is it possible the SDK sends an empty dict as the frame? Should we invalidate this case?

Reply:
It would be possible yes. Since we do frame indexing, I wouldn't be really able to remove the frame if it's invalid. We could reject the whole profile after checking if frames referenced in a sample are all valid or not though.